### PR TITLE
do not clean if packages directory is a symlink

### PR DESCRIPTION
**Pull request by stefanr**

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -947,6 +947,17 @@ function clean() {
...
+function clean() {
+  return new Promise(function(resolve, reject) {
+    fs.lstat(config.pjson.packages, function(err, stats) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(stats.isSymbolicLink() ? true : _clean());
+      }
+    });
+  });
```
Comment (guybedford) [inline lib/install.js]: We don't need to skip the entire clean, we just need to skip steps that alter `jspm_packages`, which is 4 and 5 above.

**Comment by guybedford** [comment]
Thanks for having a go at this one!

**Comment by stefanr** [comment]
Only skipping cleaning steps 4 and 5 for symlinked jspm_packages ...

**Comment by guybedford** [comment]
Amazing, thanks so much for this!

**Pull request merged**